ui/test/services/project_service_test.dart

```diff
@@ -192,6 +197,19 @@ void main() {
         expect(result.keys, ['high', 'low']);
       });
 
+      test('loads obligations', () async {
+        final dependencies = <Dependency>{};
+        dependencies.add(Dependency(id: dependencyId));
+        final obligations = {'Obligation': dependencies};
+        when(client.findObligations(projectId))
+            .thenAnswer((_) => Future.value(obligations));
+
+        final result = await service.projectObligation();
+
+        expect(result, obligations);
+        expect(result['Obligation']!.first.id, dependencyId);
```

Comment:
I suspect this lines performs a sub-check of the previous line where the total map is checked.